src/ContainerAppsResolver.ts

```diff
@@ -0,0 +1,32 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { ContainerAppsAPIClient, ManagedEnvironment } from "@azure/arm-appcontainers";
+import { callWithTelemetryAndErrorHandling, IActionContext, ISubscriptionContext } from "@microsoft/vscode-azext-utils";
+import { AppResource, AppResourceResolver, ResolvedAppResourceBase } from "@microsoft/vscode-azext-utils/hostapi";
+import { managedEnvironmentProvider } from "./constants";
+import { ResolvedContainerAppsResource } from "./tree/ResolvedContainerAppsResource";
+import { createContainerAppsAPIClient } from "./utils/azureClients";
+import { getResourceGroupFromId } from "./utils/azureUtils";
+
+export class ContainerAppsResolver implements AppResourceResolver {
+    public async resolveResource(subContext: ISubscriptionContext, resource: AppResource): Promise<ResolvedAppResourceBase | null> {
+        return await callWithTelemetryAndErrorHandling('resolveResource', async (context: IActionContext) => {
+            try {
+                const client: ContainerAppsAPIClient = await createContainerAppsAPIClient([context, subContext]);
+                const rgName: string = getResourceGroupFromId(resource.id);
+                const me: ManagedEnvironment = await client.managedEnvironments.get(rgName, resource.name);
```

Comment:
It doesn't have to be part of this PR, but if at all possible we should defer this call until it's _really_ needed, to make resolve very fast.